vms/authentication/tests/test_login.py

```diff
@@ -45,79 +46,134 @@ def tearDownClass(cls):
+        username = 'admin'
+        password = 'wrong_password'
+        self.login(username, password)
+
+        self.wait.until(
+            EC.presence_of_element_located(
+                (By.CSS_SELECTOR, '.' + AuthenticationPageLocators.INCORRECT_LOGIN_ERROR))
+        )
+
+        self.assertNotEqual(authentication_page.remove_i18n(self.driver.current_url),
+                            self.live_server_url + authentication_page.homepage)
 
-        self.assertEqual(self.driver.current_url,
+        self.assertEqual(authentication_page.remove_i18n(self.driver.current_url),
                          self.live_server_url + authentication_page.url)
 
         self.assertNotEqual(authentication_page.get_incorrect_login_message(),
                             None)
 
     def test_correct_volunteer_credentials(self):
-        '''
+        """
         Method to simulate logging in of a valid volunteer user and check if
         they are redirected to '/home'
-        '''
+        """
         authentication_page = self.authentication_page
-        self.authentication_page.server_url = self.live_server_url
-        authentication_page.login({
-            'username': 'volunteer',
-            'password': 'volunteer'
-        })
-        self.assertEqual(self.driver.current_url,
+        authentication_page.server_url = self.live_server_url
+        authentication_page.get_page(authentication_page.server_url, '')
+        username = password = 'volunteer'
+        self.login(username, password)
+
+        self.wait.until(
+            EC.presence_of_element_located(
+                (By.XPATH,
+                 "//h1[contains(text(), 'Volunteer Management System')]"))
+        )
+
+        self.assertEqual(authentication_page.remove_i18n(self.driver.current_url),
                          self.live_server_url + authentication_page.homepage)
 
-        with self.assertRaises(NoSuchElementException):
-            authentication_page.get_incorrect_login_message()
+        self.assertRaisesRegexp(NoSuchElementException,
+                                'Message: Unable to locate element: .alert-danger',
+                                authentication_page.get_incorrect_login_message)
         authentication_page.logout()
 
     def test_incorrect_volunteer_credentials(self):
-        '''
+        """
         Method to simulate logging in of a Invalid volunteer user and check if
         they are displayed an error and redirected to login page again.
-        '''
+        """
         authentication_page = self.authentication_page
-        self.authentication_page.server_url = self.live_server_url
-        authentication_page.login({
-            'username': 'volunteer',
-            'password': 'wrong_password'
-        })
+        authentication_page.server_url = self.live_server_url
+        authentication_page.get_page(authentication_page.server_url, '')
+        username = 'admin'
```

Comment:
Shouldn't you check it with `username='volunteer'`.

Reply:
Oh yes right. on it.